add start time functionality. Closes #4

johnstamatakos opened the pull request:
@marioharper Please review.

Commit:
Adds start time flag, updates commute time/date if necessary,  and adds logic for year

Commit:
Adds AM/PM support, adjusts times to local, and adjusts output labels language for current vs. future times

marioharper commented on cmd/root.go:
```diff
@@ -56,15 +59,43 @@ var RootCmd = &cobra.Command{
 		}
 
-		currTime := time.Now().Unix()
+		var commuteTime int64
```
`var commuteTime int64` is not needed if we default it to time.Now().Unix() further down

johnstamatakos replied:
👍

marioharper commented on cmd/root.go:
```diff
@@ -129,5 +160,6 @@ func init() {
 	RootCmd.PersistentFlags().StringVarP(&to, "to", "t", "home", "Destination location name")
 	RootCmd.Flags().IntVarP(&numResults, "number", "n", 5, "How many commute times do you want?")
 	RootCmd.Flags().IntVarP(&interval, "interval", "i", 15, "How many minutes between each commute prediction?")
+	RootCmd.Flags().StringVarP(&start, "start", "s", "", "Starting commute time (1219:1330 or 1219:0130PM = December 19th at 1:30PM)")
```
Should be a persistent flag so we can pass this value into the `commuter view` command. That way when they use `commuter view` it is relevant to the time they specified

johnstamatakos replied:
👍

marioharper commented on cmd/root.go:
```diff
@@ -56,15 +59,43 @@ var RootCmd = &cobra.Command{
 		var shortest int = 0
 		var optimalTime string
 		var bestHour int
+		today := true
```
determining if the date is today should be its own helper method in the Utils package not a flag manipulated by the format date logic.
```go
utils.DateIsToday(start)
```
can be called wherever, in this case on line 127 `if i == 0 && utils.DateIsToday(start) `

johnstamatakos replied:
@marioharper instead of this, I'm going to just use:
`if i == 0 && start == "" `
If the start flag is present, use a time for the first output label.  If the start flag is empty, use "Now".

Same logic without dirtying up the utils.

marioharper commented on cmd/root.go:
```diff
@@ -56,15 +59,43 @@ var RootCmd = &cobra.Command{
 		amPm := "AM"
+		commuteTime = time.Now().Unix()
+
+		if start != "" {
```
All of this logic within the if statement should be its own _formatDate helper method in this file.

```go
if start != "" {
    commuteTime = _formatDateInput(start)
}
```

johnstamatakos replied:
👍

Commit:
Fixes PR changes:

- Creates helper method for startTime logic
- Removes conditional flag
- Makes start flag persistant
- Removes unnecessary variable

marioharper commented:
sweet, just going to update the README for the new feature and then i'll merge this through

marioharper committed:
Update readme to reflect new start time feature

marioharper committed:
fix readme typo
